Library/Homebrew/dev-cmd/edit.rb

```diff
@@ -7,6 +7,32 @@
 module Homebrew
   module_function
 
+  module EditPathnameRefinements
+    refine Pathname do
+      sig { returns(T::Boolean) }
+      def core_formula_path?
+        fnmatch?("**/homebrew-core/Formula/**.rb", File::FNM_DOTMATCH)
+      end
+
+      sig { returns(T::Boolean) }
+      def core_cask_path?
+        fnmatch?("**/homebrew-cask/Casks/**.rb", File::FNM_DOTMATCH)
+      end
+
+      sig { returns(T::Boolean) }
+      def core_formula_tap?
+        self == CoreTap.instance.path
+      end
+
+      sig { returns(T::Boolean) }
+      def core_cask_tap?
+        self == CoreCaskTap.instance.path
+      end
+    end
+  end
+
+  using EditPathnameRefinements
```

Comment:
I think just remove this refinement, don't add to `Pathname` and instead turn these methods into e.g. 
```ruby
sig { params(path: Pathname).returns(T::Boolean) }
def core_formula_path?(path)
```

Reply:
Done.